applications/tari_console_wallet/src/main.rs

```diff
@@ -43,17 +43,23 @@ mod utils;
 
 /// Application entry point
 fn main() {
-    match main_inner() {
-        Ok(_) => std::process::exit(0),
+    let mut shutdown = Shutdown::new();
+
+    match main_inner(shutdown.to_signal()) {
+        Ok(_) => {
+            shutdown.trigger().unwrap();
```

Comment:
After we trigger the shutdown we need to wait on the wallet to complete its shutdown (which really just waits for the comms stack to finish its shutdown). 

I recommend we change the wallets shutdown (copied below) to a wait_until_shutdown.

```Rust
pub async fn shutdown(mut self) {
        self.shutdown.trigger().expect(
            "No one is listening for shutdown trigger. If you are seeing this error then the shutdown signals have \
             not been wired up correctly.",
        );
        self.comms.wait_until_shutdown().await;
    }
```

Also whenever you are using an unwrap it is better to use an .expect("...") with a message saying what was supposed to happen.

Reply:
Excellent thanks will do